dns_server/src/main.rs

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+
+use clap::Parser;
+use config::DnsServerConfig;
+use crawler::{storage_impl::DnsServerStorageImpl, Crawler};
+use p2p::net::default_backend::{
+    transport::{NoiseEncryptionAdapter, NoiseTcpTransport, TcpTransportSocket},
+    DefaultNetworkingService,
+};
+use tokio::sync::mpsc;
+
+mod config;
+mod crawler;
+mod dns_server;
+mod error;
+
+fn make_p2p_transport() -> NoiseTcpTransport {
```

Comment:
I think this can carry risk. The decision of transport should be made in p2p and exported here. This is code duplication.

The bad thing with this is that in the future, if we change the wrapping protocol, this will break the DNS server.

Reply:
Updated, thanks.